src/app/pages/send/send-crypto-asset-preview/hooks/use-send-form-details-and-action.tsx

```diff
@@ -0,0 +1,58 @@
+import { CryptoCurrencies } from '@shared/models/currencies.model';
+
+import { SendCryptoAssetButton } from '../components/send-crypto-asset-button';
+import { StxSendFormDetails } from '../stx/stx-send-form-details';
+import { useStxBroadcastTransaction } from '../stx/use-stx-broadcast-transaction';
+
+interface UseSendFormDetailsAndActionArgs {
+  feeCurrency: CryptoCurrencies;
+  symbol: string;
+  token: string;
+  tx: string;
+}
+export function useSendFormDetailsAndAction({
+  feeCurrency,
+  symbol,
+  token,
+  tx,
+}: UseSendFormDetailsAndActionArgs) {
```

Comment:
Can't this be a component? Don't understand why it's a hook.

Pseudoish code

```ts
function SendFormDetailsAndAction (props) {
  const { stxDeserializedTransaction, stxBroadcastTransaction } = useStxBroadcastTransaction(tx);
  switch () {
    case 'stx': {
      return <Compoent />
    }
  }
}  
```

Reply:
Yeah, not sure why I did that. Will change, thx.